File: Lexer.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
/**
 * This class is thread safe.
 */
public class Lexer {
  private File input;
  public synchronized void setFile(File f) {
    input = f;
  }
  public synchronized File file() {
    return input;
  }
  public String readContent() throws IOException {
    FileInputStream i = new FileInputStream(input);
    String t = "";
    int data;
    while ((data = i.read()) > 0) {
      t += (char) data;
    }
    return t;
  }
  public String readContentWOUnicode() throws IOException {
    FileInputStream i = new FileInputStream(input);
    String temp = "";
    int data;
    while ((data = i.read()) > 0) {
      if (data < 0x80) {
        temp += (char) data;
      }
    }
    return temp;
  }
  public void saveContent(String text) {
    FileOutputStream o = new FileOutputStream(input);
    try {
      for (int i = 0; i < text.length(); i += 1) {
        o.write(text.charAt(i));
      }
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}

File: Lexer.kt
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException

/**
 * Этот класс потокобезопасен.
 */
class Lexer {
    private var input: File? = null

    @Synchronized
    fun setFile(f: File) {
        input = f
    }

    @Synchronized
    fun file(): File? {
        return input
    }

    @Throws(IOException::class)
    fun readContent(): String {
        val i = FileInputStream(input)
        var content = ""
        var data: Int
        while (i.read().also { data = it } > 0) {
            content += data.toChar()
        }
        return content
    }

    @Throws(IOException::class)
    fun readContentWOUnicode(): String {
        val i = FileInputStream(input)
        var temp = ""
        var data: Int
        while (i.read().also { data = it } > 0) {
            if (data < 0x80) {
                temp += data.toChar()
            }
        }
        return temp
    }

    fun saveContent(text: String) {
        val o = FileOutputStream(input)
        try {
            for (char in text) {
                o.write(char.toInt())
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}
